src/swap/types.ts

```diff
@@ -116,18 +127,28 @@ export type SwapContextType = {
   error?: SwapError;
   fromAmount: string;
   fromToken?: Token;
+  fromTokenBalance?: string;
+  isLoading?: SwapLoadingState;
   handleFromAmountChange: (a: string) => void;
   handleToAmountChange: (a: string) => void;
   handleToggle: () => void;
   setError: (e: SwapError) => void;
   setFromAmount: (a: string) => void;
   setFromToken: (t: Token) => void;
+  setFromTokenBalance: (b: string) => void;
+  setIsLoading: (l: SwapLoadingState) => void;
   setToAmount: (a: string) => void;
   setToToken: (t: Token) => void;
   toAmount: string;
   toToken?: Token;
 };
 
+export type SwapLoadingState = {
+  fromTokenBalance?: boolean;
+  swapQuote?: boolean;
```

Comment:
When working with booleans, it's good practice to either start the variable with `is` or `has`.

So when reading the code it's easier to know the potential type without always triple checking.